### ISSUE TITLE
How to inject PoW?

### ISSUE THREAD

**Issue by kilkelly** [marvinroger/nanocurrency-js#9]
In the documentation when creating blocks it states:

> You will have to inject the PoW.

however I can't find any example of how to do this. Did I miss an example somewhere? Thanks!

**Comment by marvinroger** [comment]
A block looks like that: 

```json
{
    "type": "open",
    "source": "19D3D919475DEED4696B5D13018151D1AF88B2BD3BCFF048B45031C1F36D1858",
    "representative": "xrb_1hza3f7wiiqa7ig3jczyxj5yo86yegcmqk3criaz838j91sxcckpfhbhhra1",
    "account": "xrb_3kdbxitaj7f6mrir6miiwtw4muhcc58e6tn5st6rfaxsdnb7gr4roudwn951",
    "work": "4ec76c9bda2325ed",
    "signature": "5974324F8CC42DA56F62FC212A17886BDCB18DE363D04DA84EEDC99CB4A33919D14A2CF9DE9D534FAA6D0B91D01F0622205D898293525E692586C84F2DCF9208"
}
```

The `createOpenBlock` function of the lib would return that:

```json
{
    "type": "open",
    "source": "19D3D919475DEED4696B5D13018151D1AF88B2BD3BCFF048B45031C1F36D1858",
    "representative": "xrb_1hza3f7wiiqa7ig3jczyxj5yo86yegcmqk3criaz838j91sxcckpfhbhhra1",
    "account": "xrb_3kdbxitaj7f6mrir6miiwtw4muhcc58e6tn5st6rfaxsdnb7gr4roudwn951",
    "work": null,
    "signature": "5974324F8CC42DA56F62FC212A17886BDCB18DE363D04DA84EEDC99CB4A33919D14A2CF9DE9D534FAA6D0B91D01F0622205D898293525E692586C84F2DCF9208"
}
```

"injecting" the PoW simply means that you'll need to fill `work`yourself (from a pre-computed work, a remote work generation or from a `work`call)

**Comment by kilkelly** [comment]
Thanks! I was guessing that it could be like that after looking harder. And the work is computed using the hash of the newly created block?

**Comment by marvinroger** [comment]
No, using the hash of the previous block on the account chain. That way, you can precompute the work for an eventual next send or receive block, that would therefore be instant (the work being precomputed and cached).

**Comment by marvinroger** [comment]
Note, if there’s no previous block (that’s the case when you open an account with an open block), you have to use the account public key instead of the hash.

**Comment by kilkelly** [comment]
Looks like I completely forgot about how a blockchain works there for a second :D Thanks for clarifying, it makes perfect sense now.

**Comment by marvinroger** [comment]
No problem :)
Actually, you made me realize that, as the work only needs the previous block, there's no need to inject the PoW after creating it, the methods can have a "work" parameter, it would make more sense!

**Comment by kilkelly** [comment]
Yes, that would be nice to have as an optional parameter 👍 

I seem to be struggling to understand the origin of the 'source hash' required in the create* methods of the API. Could you clarify how to get this, especially with regards to the `createOpenBlock` method? I can open an new issue if this doesn't belong here. Thanks!

**Comment by marvinroger** [comment]
I am currently rewriting the docs, here's the  new info:

> The hash of the send block that is being received, in hexadecimal format

You can only open an account (e.g send an open block) when you've sent a send block to it from another account. Take a look at that: https://www.nanode.co/docs

> Opening an account
An account's first transaction must always be an open block. An open block is like a receive block--it must have an accompanying send transaction in order to be valid. In other words, you need to send Nano to an account's address before you can open it.

**Comment by kilkelly** [comment]
Makes complete sense, thank you @marvinroger

**Comment by marvinroger** [comment]
I just released the 1.7.0 😉 
The parameters should be easier to understand, also. Note there are no tests yet for the block creation methods, so it might be broken somehow. Don't hesitate to report if the blocks are not valid.

I'll add tests, anyway!
